lib/dao/dao_factory.es6.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.INTERFACE({
```

Comment:
It surprises me that passing in a DAO's class or type is not sufficient.

Does FOAM have a dependency injection framework, or equivalent?

Reply:
Yes, we could use a class. My intention here was to make the configuration of underlying classes more explicit. E.g., we could have something like:

```js
foam.CLASS({
  name: 'CompatRestDAO',
  extends: 'foam.dao.RestDAO',

  properties: [
    {name: 'baseURL', value: 'https://web-confluence.appspot.com/compatDAO'},
  ],
});

.CLASS({
  name: 'WebInterfaceRestDAO',
  extends: 'foam.dao.RestDAO',

  properties: [
    {name: 'baseURL', value: 'https://web-confluence.appspot.com/webInterfaceDAO'},
  ],
});
/* ... */
```

I thought appropriately configured `RestDAOFactory` instances made more sense, but I don't have a strong feeling. we can just as easily use `remotDAOCls` and adopt the above pattern. Do you think using classes instead of factories is simpler/easier to understand/(better in some way I haven't thought of)?